core/chains/evm/client/client.go

```diff
@@ -185,8 +168,12 @@ func (client *client) TransactionReceipt(ctx context.Context, txHash common.Hash
 	return
 }
 
-func (client *client) ChainID() *big.Int {
-	return client.pool.chainID
+func (client *client) TransactionByHash(ctx context.Context, txHash common.Hash) (tx *types.Transaction, err error) {
+	return client.pool.TransactionByHash(ctx, txHash)
+}
+
+func (client *client) ChainID() (*big.Int, error) {
+	return client.pool.chainID, nil
```

Comment:
Unless we plan on changing the `chainID` from a configuration to an RPC call, I don't see a reason why this function should return an error value.

Reply:
It's an RPC/Gateway call on other non-evm chains ([example](https://github.com/smartcontractkit/chainlink-starknet/blob/92ef75c9de2041b9969b0a36d627ccc2c2df8a2f/relayer/pkg/starknet/client.go#L192)). Also I think that as a chain interface it should describe the way to fetch `chainId` from chain rather than from configuration.

Reply:
Until recently we made the assumption that chain id was already known to spin up a node (more specifically to apply chain-specific configurations). I'm not aware, but we still might expect that in other parts of our code. 
Overall getting the chain id from configs is much more efficient than making a request each time. ETH also supports getting the chain id via API call but it's much simpler to add it when setting up the node. My suggestion would be to either continue under the assumption that chain id should be known beforehand or turn this method into an API call and cache the result t avoid unnecessary calls.

Reply:
imo we should validate the chain ID of the nodes in the pool against the configured chain ID, but that might be better as a separate ticket?

if we validate during start up, I think we don't need to return an error with this call since then it's just cached?

also not having an error for this call would reduce the footprint of this PR